code/modules/surgery/organs/liver.dm

```diff
@@ -119,25 +119,26 @@
 
 /obj/item/organ/internal/liver/organ_failure(delta_time)
 
-	switch(failure_time/LIVER_FAILURE_STAGE_SECONDS)
-		if(1)
-			to_chat(owner, span_userdanger("You feel stabbing pain in your abdomen!"))
-		if(2)
-			to_chat(owner, span_userdanger("You feel a burning sensation in your gut!"))
-			owner.vomit()
-		if(3)
-			to_chat(owner, span_userdanger("You feel painful acid in your throat!"))
-			owner.vomit(blood = TRUE)
-		if(4)
-			to_chat(owner, span_userdanger("Overwhelming pain knocks you out!"))
-			owner.vomit(blood = TRUE, distance = rand(1,2))
-			owner.emote("Scream")
-			owner.AdjustUnconscious(2.5 SECONDS)
-		if(5)
-			to_chat(owner, span_userdanger("You feel as if your guts are about to melt!"))
-			owner.vomit(blood = TRUE,distance = rand(1,3))
-			owner.emote("Scream")
-			owner.AdjustUnconscious(5 SECONDS)
+	if(!HAS_TRAIT(owner, TRAIT_NO_PAIN))
```

Comment:
Even without pain, liver failure should still cause you to vomit.

Reply:
I have changed this so the vomiting still happens, but the other effects do not, if you have this quirk.